src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -0,0 +1,91 @@
+package com.commercetools.emailprocessor.email;
+
+import com.commercetools.emailprocessor.model.Email;
+import com.commercetools.emailprocessor.model.Statistics;
+import com.commercetools.emailprocessor.model.TenantConfiguration;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+
+public class EmailProcessor {
+
+
+    private static final Logger LOG = LoggerFactory.getLogger(EmailProcessor.class);
+    private static final String PARAM_EMAIL_ID = "emailid";
+    private static final String PARAM_TENANT_ID = "tenantid";
+    private static final String CONTAINER_ID = "unprocessedEmail";
+  ;
+    private static final String EMAIL_STATUS_PENDING = "pending";
+
+    public EmailProcessor() {
+    }
+
+    /**
+     * Iterate through all Email objects and triggers the webhook for each  pending email object
+     *
+     * @param tenantConfiguration Configuration of a tenant
+     * @return Statics of the sended emails
+     */
+
+    public Statistics processEmails(TenantConfiguration tenantConfiguration) {
```

Comment:
An integration test for this would be good. You can create a fake custom object, which will have a webhook e.g. https://httpbin.org/, then run the thing, and see if you have correct statistics.

Reply:
d2d07db

Reply:
Pls test `Main.main()`. You can start a new webserver https://stackoverflow.com/questions/29758607/how-to-run-jetty-server-for-java-junit-testing. If it's too much job, just create an issue.

Reply:
https://github.com/commercetools/commercetools-email-retry-processor/issues/3